pages/search/advanced.tsx

```diff
@@ -0,0 +1,225 @@
+    ).key
+
+    dispatch({
+      type: type,
+      field: searchParamsKey,
+      payload: target.value,
+    })
+  }
+
+  const handleDateChange = (field: string, e: FullDateType) => {
+    alert && setAlert(false)
+    const yearString = e.startDate?.getFullYear()?.toString()
+
+    dispatch({
+      type: "filter_change",
+      field: field,
+      payload: yearString || "",
+    })
+  }
+
+  const handleCheckboxChange = (types: string[]) => {
+    alert && setAlert(false)
+    dispatch({
+      type: "filter_change",
+      field: "materialType",
+      payload: types,
+    })
+  }
+
+  const handleSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault()
+
+    const queryString = getQueryString(searchFormState as SearchParams)
+    if (!queryString.length) {
+      setAlert(true)
+    } else {
+      await router.push(`/search/?${queryString}`)
+    }
+  }
+
+  const handleClear = (e: SyntheticEvent) => {
+    e.preventDefault()
+    dispatch({ type: "form_reset", payload: initialSearchFormState })
+  }
+
+  return (
+    <>
+      <Head>
+        <title>Advanced Search | {SITE_NAME}</title>
+      </Head>
+      {alert && (
+        <Notification
+          notificationType="warning"
+          notificationContent={
+            <>Please enter at least one field to submit an advanced search.</>
+          }
+        />
+      )}
+      <Heading level="two">Advanced Search</Heading>
+      <Form
+        id="advancedSearchForm"
+        method="post"
+        action={`${BASE_URL}/search`}
+        onSubmit={handleSubmit}
+      >
+        <FormRow gap="grid.m">
+          <FormField id="advancedSearchLeft" gap="grid.s">
+            {textInputFields.map(({ name, key, label }) => {
+              return (
+                <TextInput
+                  id={name}
+                  labelText={label}
+                  type="text"
+                  name={name}
+                  value={searchFormState[name]}
+                  key={key}
+                  onChange={(e) => handleInputChange(e, "input_change")}
+                />
+              )
+            })}
+            <Select
+              id="languageSelect"
+              name="language"
+              labelText="Language"
+              aria-labelledby="languageSelect-label"
+              value={searchFormState["selectedFilters"].language}
+              onChange={(e) => handleInputChange(e, "filter_change")}
+            >
+              {languageOptions.map((language) => {
+                return (
+                  <option value={language.value} key={language.value}>
+                    {language.label}
+                  </option>
+                )
+              })}
+            </Select>
+          </FormField>
+          <FormField id="advancedSearchRight" gap="grid.s">
+            <Fieldset legendText="Date" id="dates">
+              <FormRow id="dates" gap="grid.m">
+                <FormField>
+                  <DatePicker
```

Comment:
Ah. I was wondering how `handleDateChange` was being used and just realized these are two DatePickers. Did you try making it a single DatePicker as a range type and find out that it didn't work for this situation?

Reply:
No but I assumed that users might want to only select one date, rather than a range. Not sure if this is possible with a date range?

Reply:
hmm true. that's a good product question. Leave it as  is for now and we should get more info on this.